src/libmongoc/src/mongoc/mongoc-topology-scanner.c

```diff
@@ -104,9 +104,22 @@ _jumpstart_other_acmds (mongoc_topology_scanner_node_t *node,
                         mongoc_async_cmd_t *acmd);
 
 static void
-_add_ismaster (bson_t *cmd)
+_add_ismaster (bson_t *cmd, const mongoc_server_api_t *api)
 {
    BSON_APPEND_INT32 (cmd, "isMaster", 1);
+
+   if (api) {
+      _mongoc_cmd_append_server_api (cmd, api);
+   }
+}
+
+static void
+_reset_ismaster (mongoc_topology_scanner_t *ts)
+{
+   bson_init (&ts->ismaster_cmd);
```

Comment:
Yes, this may have leaked memory. `bson_reinit` only works on a `bson_t` that was previously initialised, so I've split up the method. I've also realised that we shouldn't clear the `cluster_time`, so I've removed that from `_reset_ismaster` as well.